### ISSUE TITLE
 hide BottomNavigatorsBar

### ISSUE THREAD

**Issue by sh4msi** [zenonine/navi#32]
hi,
What should I do if I want to hide BottomNavigatorsBar on a specific page when using nested navigator?

**Comment by xuanswe** [comment]
Hi @sh4msi,

Thanks for trying Navi!

The hiding of BottomNavigationBar is not controlled by Navi. But if the hiding logic depends on the state of the nested navigator, you could listen to the current RouteStack state (maybe in your `initState()`) to show/hide the bar when state is changed.

```
final _stackController = StackController();

void initState() {
// ...
  _stackController.addListener(() {
    // show/hide BottomNavigationBar based on _stackController.state
  });
}

// ...
build() {
  // ...
    RouteStack(
      controller: _stackController,
    )
  // ...
}
```

If it's still unclear, it's better if you could provide your simplified version of your code.

Note that, current version is not yet tested carefully for complex scenario. So many things will still need to be improved to adapt real application.

**Comment by xuanswe** [comment]
FYI, because using tabs is a very common scenario. I am planning to make it easier to use. The change would be a wrapper widget for nested route, which excepts a controller to update itself. This widget will be general enough to work not just with `BottomNavigationBar` but any similar components or designs (ex. `TabBar` / `TabBarView`).

**Comment by xuanswe** [comment]
@sh4msi I released 0.2.0-rc+0, updated documentation, add more [examples](https://github.com/zenonine/navi/tree/master/examples). It works much more stable now.

I decide not to create wrapper widget because the API in new version is so simple to use in even complex nested routes.
